datajunction-server/datajunction_server/materialization/jobs/cube_materialization.py

```diff
@@ -82,46 +87,42 @@ def build_druid_spec(self, cube_config: DruidCubeConfig, node_name: str) -> Dict
         }
 
         metrics_spec = list(_metrics_spec.values())
-        temporal_partitions = (
-            [
-                partition
-                for partition in cube_config.partitions
-                if partition.type_ == PartitionType.TEMPORAL
-            ]
-            if cube_config.partitions
-            else []
-        )
-        if not temporal_partitions:
-            raise DJException(
-                "Druid ingestion requires a temporal partition to be specified",
+        granularity, timestamp_column = None, None
+        for col in node_revision.columns:
+            if col.partition and col.partition.type_ == PartitionType.TEMPORAL:
+                granularity = "DAY"
```

Comment:
Does this mean temporal partitions on cubes are always daily?

Reply:
Good point -- I just updated this to add support for users providing a datetime format (like `yyyyMMdd`) + granularity and DJ building out the expected expression/intervals based on that.